ext/com/concurrent_ruby/ext/SynchronizationLibrary.java

```diff
@@ -55,14 +55,14 @@ public IRubyObject initialize(ThreadContext context, IRubyObject[] args, Block b
             }
         }
 
-        @JRubyMethod(name = "synchronize", visibility = Visibility.PRIVATE)
+        @JRubyMethod(name = "synchronize", visibility = Visibility.PROTECTED)
```

Comment:
@pitr-ch All these years using Ruby and I never realize that subclasses could access private methods of superclasses--simply because I've never tried it. Your phrase "counter intuitive" is very polite. I like the word "dumb" better. That's not what `protected` is supposed to mean! I'll bring that up with Matz the next time he and I chat. :-) Seriously, though, I had to write some code and run it myself. Sadly, it worked exactly as you described...

@headius Ruby is as Ruby does, I guess. Thanks for taking the time to look into it but for now we'll just keep it protected. Behavior aside, I think `protected` is semantically and logically correct for what we're doing.